windows/devices/uwb/UwbCxAdapterDdiLrp.cxx

```diff
@@ -534,10 +534,95 @@ windows::devices::uwb::ddi::lrp::To(const UWB_DEVICE_CAPABILITIES &deviceCapabil
+    { UWB_STATUS_RANGING_RX_PHY_STS_FAILED, UwbStatusRanging::RxPhyStsFailed },
+    { UWB_STATUS_RANGING_RX_MAC_DEC_FAILED, UwbStatusRanging::MacDecodingFailed },
+    { UWB_STATUS_RANGING_RX_MAC_IE_DEC_FAILED, UwbStatusRanging::RxMacIeDecodingFailed },
+    { UWB_STATUS_RANGING_RX_MAC_IE_MISSING, UwbStatusRanging::RxMacIeMissing },
+};
+
+static const std::unordered_map<UWB_SESSION_STATE, UwbSessionState> SessionStateToMap{
+    { UWB_SESSION_STATE_INIT, UwbSessionState::Initialized },
+    { UWB_SESSION_STATE_DEINIT, UwbSessionState::Deinitialized },
+    { UWB_SESSION_STATE_ACTIVE, UwbSessionState::Active },
+    { UWB_SESSION_STATE_IDLE, UwbSessionState::Idle },
+};
+
+static const std::unordered_map<UWB_SESSION_REASON_CODE, UwbSessionReasonCode> SessionReasonCodeToMap{
+    { UWB_SESSION_REASON_CODE_STATE_CHANGE_WITH_SESSION_MANAGEMENT_COMMANDS, UwbSessionReasonCode::StateChangeWithSessionManagementCommands },
+    { UWB_SESSION_REASON_CODE_MAX_RANGING_ROUND_RETRY_COUNT_REACHED, UwbSessionReasonCode::MaxRangignRoundRetryCountReached },
+    { UWB_SESSION_REASON_CODE_MAX_NUMBER_OF_MEASUREMENTS_REACHED, UwbSessionReasonCode::MaxNumberOfMeasurementsReached },
+    { UWB_SESSION_REASON_CODE_ERROR_SLOT_LENGTH_NOT_SUPPORTED, UwbSessionReasonCode::ErrorSlotLengthNotSupported },
+    { UWB_SESSION_REASON_CODE_ERROR_INSUFFICIENT_SLOTS_PER_RR, UwbSessionReasonCode::ErrorInsufficientSlotsPerRangingRound },
+    { UWB_SESSION_REASON_CODE_ERROR_MAC_ADDRESS_MODE_NOT_SUPPORTED, UwbSessionReasonCode::ErrorMacAddressModeNotSupported },
+    { UWB_SESSION_REASON_CODE_ERROR_INVALID_RANGING_INTERVAL, UwbSessionReasonCode::ErrorInvalidRangingInterval },
+    { UWB_SESSION_REASON_CODE_ERROR_INVALID_STS_CONFIG, UwbSessionReasonCode::ErrorInvalidStsConfiguration },
+    { UWB_SESSION_REASON_CODE_ERROR_INVALID_RFRAME_CONFIG, UwbSessionReasonCode::ErrorInvalidRFrameConfiguration },
+};
+
 UwbNotificationData
-windows::devices::uwb::ddi::lrp::To(const UWB_NOTIFICATION_DATA & /* notificationData */)
+windows::devices::uwb::ddi::lrp::To(const UWB_NOTIFICATION_DATA &notificationData)
 {
-    UwbNotificationData uwbNotificationData{};
-    // TODO: implement this
-    return uwbNotificationData;
+    // TODO: finish implementing this
+    switch (notificationData.notificationType) {
+    case UWB_NOTIFICATION_TYPE_DEVICE_STATUS: {
+        return UwbStatusDevice{ .State = DeviceStateToMap.at(notificationData.deviceStatus.deviceState) };
+    }
+    case UWB_NOTIFICATION_TYPE_GENERIC_ERROR: {
+        auto enumId = notstd::to_underlying(notificationData.genericError);
+        if (enumId < notstd::to_underlying(UWB_STATUS_ERROR_SESSION_NOT_EXIST)) {
+            return StatusToMapGeneric.at(notificationData.genericError);
+        }
+        if (enumId < notstd::to_underlying(UWB_STATUS_RANGING_TX_FAILED)) {
+            return StatusToMapSession.at(notificationData.genericError);
+        }
+        return StatusToMapRanging.at(notificationData.genericError);

```

Comment:
We don't need a special case for this. We should just pass `notificationData.genericError` to the `::To(UWB_STATUS& status)` function which returns a `UwbStatus` (I suggested writing this function in another comment).